### PR TITLE
Updated encryption examples

### PR DESCRIPTION
**Pull request by martinthomson**
Thanks to @beverloo for identifying the bug in my code that caused this to take so long.

### CODE REVIEW

File: draft-ietf-httpbis-encryption-encoding.md [hunk 1]
```diff
@@ -598,6 +629,27 @@ added for presentation purposes only.
...
+~~~
+HTTP/1.1 200 OK
+Content-Length: 33
+Content-Encoding: aesgcm
+Encryption: keyid="dhkey"; salt="lngarbyKfMoi9Z75xYXmkg"
+Crypto-Key: keyid="dhkey";
+                dh="BNoRDbb84JGm8g5Z5CFxurSqsXWJ11ItfXEWYVLE85Y7
+                    CYkDjXsIEc4aqxYaQ1G8BqkXCJ6DPpDrWtdWj_mugHU"
+
+QPsWCEH-8WqiXl7Ez23A6u4rwH2hFBifCybQQz-SxsjF
```
Comment (beverloo) [inline draft-ietf-httpbis-encryption-encoding.md]: I am unable to reproduce this in either of my implementations, using the following information:

**Sender public**: BNoRDbb84JGm8g5Z5CFxurSqsXWJ11ItfXEWYVLE85Y7CYkDjXsIEc4aqxYaQ1G8BqkXCJ6DPpDrWtdWj_mugHU
**Sender private**: nCScek-QpEjmOOlT-rQ38nZzvdPlqa00Zy0i6m2OJvY
**Receiver public**: BCEkBjzL8Z3C-oi2Q7oE5t2Np-p7osjGLg93qUP0wvqRT21EEWyf0cQDQcakQMqz4hQKYOQ3il2nNZct4HgAUQU
**Salt**: lngarbyKfMoi9Z75xYXmkg
**Auth secret**: R29vIGdvbyBnJyBqb29iIQ
**Padding**: 0 bytes
**Plaintext**: I am the walrus

I get the following:

**Shared secret**: RNjC-NVW4BGJbxWPW7G2mowsLeDa53LYKYm4--NOQ6Y
**PRK**: EhpZec37Ptm4IRD5-jtZ0q6r1iK5vYmY1tZwtN8fbZY
**CEK**: AN2-xhvFWeYh5z0fcDu0Ww
**Nonce**: JY1Okw5rw1Drkg9J
**Ciphertext**: 6nqAQUME8hNqw5J3kl8cpVVJylXKYqZOeseZG8UueKpA

Decryption fails as well, so I can't see whether you used a different `plaintext`.

File: draft-ietf-httpbis-encryption-encoding.md [hunk 2]
```diff
@@ -359,13 +359,13 @@ header field.
 
 aesgcm:
 
-: The "aesgcm" parameter contains the URL-safe base64 [RFC4648] octets of the
-input keying material.
+: The "aesgcm" parameter contains the URL and filename safe base64 [RFC4648]
+  octets of the input keying material with padding removed.
```
Comment (beverloo) [inline draft-ietf-httpbis-encryption-encoding.md]: Do you want to be strict in regards to omitting padding? Right now our implementation is lenient and just ignores it.

File: draft-ietf-httpbis-encryption-encoding.md (hunk 1 above)
Comment (martinthomson) [inline draft-ietf-httpbis-encryption-encoding.md]: Ahh, I get the same answer that you do.  Strange.  I must have I copied the wrong value in somehow.  Thanks for checking!  I'll change the ciphertext.

File: draft-ietf-httpbis-encryption-encoding.md (hunk 2 above)
Comment (martinthomson) [inline draft-ietf-httpbis-encryption-encoding.md]: Well, it turns out that '=' is an invalid character unless you quote the string.  I think that we should be strict here.  I believe that Firefox is, but I could be wrong about that.

Comment (beverloo) [inline draft-ietf-httpbis-encryption-encoding.md]: I'll change Chrome to play ball then.

File: draft-ietf-httpbis-encryption-encoding.md [hunk 3]
```diff
@@ -598,6 +629,28 @@ added for presentation purposes only.
...
+~~~
+HTTP/1.1 200 OK
+Content-Length: 33
+Content-Encoding: aesgcm
+Encryption: keyid="dhkey"; salt="lngarbyKfMoi9Z75xYXmkg"
+Crypto-Key: keyid="dhkey";
+                dh="BNoRDbb84JGm8g5Z5CFxurSqsXWJ11ItfXEWYVLE85Y7
+                    CYkDjXsIEc4aqxYaQ1G8BqkXCJ6DPpDrWtdWj_mugHU"
+
+6nqAQUME8hNqw5J3kl8cpVVJylXKYqZOeseZG8UueKpA
```
Comment (beverloo) [inline draft-ietf-httpbis-encryption-encoding.md]: Confirmed! Thanks.

**Comment by martinthomson** [comment]
Merged manually

**Pull request closed without merging**